Security.md

```diff
@@ -0,0 +1,123 @@
+# Security
+
+The security model of WebAssembly has two important goals: (1) protect *users*
+from buggy or malicious modules, and (2) provide *developers* with useful
+primitives and mitigations for developing safe applications, within the
+constraints of (1).
+
+## Users
+
+Each WebAssembly module executes within a sandboxed environment separated from
+the host runtime using fault isolation techniques. This implies:
+  * Applications execute independently (unless
+  [dynamic linking](DynamicLinking.md) is used), and can't escape the sandbox
+  without going through appropriate APIs (the reference monitor).
+  * Applications generally execute deterministically
+  [with limited exceptions](Nondeterminism.md).
+
+Additionally, each module is subject to the security policies of its embedding.
+Within a [web browser](Web.md), this includes restrictions on information flow
+through [same-origin policy][]. On a [non-web](NonWeb.md) platform, this could
+include the POSIX security model.
+
+## Developers
+
+The design of WebAssembly promotes safe programs by eliminating dangerous
+features from its execution semantics, while maintaining compatibility with
+programs written for [C/C++](CandC++.md).
+
+Modules must declare all accessible functions and their associated types
+at load time, even when [dynamic linking](DynamicLinking.md) is used. This
+allows implicit enforcement of [control-flow integrity][] (CFI) through
+structured control-flow. Since compiled code is immutable and not observable at
+runtime, WebAssembly programs are protected from control flow hijacking attacks.
+  * [Function calls](AstSemantics.md#calls) must specify the index of a target
+  that corresponds to a valid entry in the
+  [function index space](Modules.md#function-index-space) or
+  [table index space](Modules.md#table-index-space).
+  * [Indirect function calls](Rationale.md#indirect-calls) are subject to a type
+  signature check at runtime; the type signature of the selected indirect
+  function must match the type signature specified at the call site.
+  * A shadow stack is used to maintain a trusted call stack that is invulnerable
+  to buffer overflows in the module heap, ensuring safe function returns.
+  * [Branches](AstSemantics.md#branches-and-nesting) must point to valid
+  destinations within the enclosing function.
+
+Variables in C/C++ can be lowered to two different primitives in WebAssembly,
+depending on their scope. [Local variables](AstSemantics.md#local-variables)
+with fixed scope and [global variables](AstSemantics.md#global-variables) are
```

Comment:
Wasm doesn't have globals like asm.js.

... Come to think of it, why is that section still in AstSemantics.md, shouldn't it be removed? (Not for this PR)